GherkinSyncTool.Synchronizers.AllureTestOps/Content/CaseContentBuilder.cs

```diff
@@ -45,19 +47,125 @@ public CaseContentBuilder(AllureClientWrapper allureClientWrapper, Context conte
+                Scenario = AddScenario(scenario, featureFile)
+            },
+            StepsAttachments = AddStepAttachments(scenario, featureFile)
+        };
+
+        return createTestCaseRequestExtended;
+    }
+
+    private Dictionary<int, ByteArrayPart> AddStepAttachments(Scenario scenario, IFeatureFile featureFile)
+    {
+        var extractAttachments = ExtractAttachments(scenario.Steps.ToList());
+
+        var background = featureFile.Document.Feature.Children.OfType<Background>().FirstOrDefault();
+        if (background is not null)
+        {
+            var backgroundAttachments = ExtractAttachments(background.Steps.ToList());
+            return backgroundAttachments.Concat(extractAttachments).ToDictionary(pair => pair.Key, pair => pair.Value);
+        }
+
+        return extractAttachments;
+    }
+
+    private Dictionary<int, ByteArrayPart> ExtractAttachments(List<Step> steps)
+    {
+        var result = new Dictionary<int, ByteArrayPart>();
+
+        for (var index = 0; index < steps.Count; index++)
+        {
+            var step = steps[index];
+
+            switch (step.Argument)
+            {
+                case DocString docString:
+                    
+                    var textBytes = Encoding.ASCII.GetBytes(docString.Content);
+                    result.Add(index, new ByteArrayPart(textBytes, "Text", "text/plain"));
+                    break;
+                
+                case DataTable table:
+                    
+                    var csvBytes = Encoding.ASCII.GetBytes(ConvertToCsvTable(table.Rows.ToList()));
+                    result.Add(index, new ByteArrayPart(csvBytes, "Table", "text/csv"));
+                    break;
+            }
+        }
+
+        return result;
+    }
+
+    private Quantori.AllureTestOpsClient.Model.Scenario AddScenario(Scenario scenario, IFeatureFile featureFile)
+    {
+        var steps = GetSteps(scenario, featureFile);
+        if (!steps.Any()) return null;
+
+        var allureScenario = new Quantori.AllureTestOpsClient.Model.Scenario
+        {
+            Steps = new List<Quantori.AllureTestOpsClient.Model.Step>()
         };
+        allureScenario.Steps.AddRange(steps);
+        return allureScenario;
+    }
+
+    private List<Quantori.AllureTestOpsClient.Model.Step> GetSteps(Scenario scenario, IFeatureFile featureFile)
+    {
+        var scenarioSteps = ExtractSteps(scenario.Steps.ToList());
 
-        return caseRequest;
+        var background = featureFile.Document.Feature.Children.OfType<Background>().FirstOrDefault();
+        if (background is not null)
+        {
+            var backgroundSteps = ExtractSteps(background.Steps.ToList());
+            return backgroundSteps.Concat(scenarioSteps).ToList();
+        }
+
+        return scenarioSteps;
+    }
+
+    private List<Quantori.AllureTestOpsClient.Model.Step> ExtractSteps(List<Step> steps)
+    {
+        var allureSteps = new List<Quantori.AllureTestOpsClient.Model.Step>();
```

Comment:
You can implement this to make code more readable:
`using AllureTestCaseStep = Quantori.AllureTestOpsClient.Model.Step` in the usings block

Reply:
Done